specification/search/data-plane/Microsoft.Azure.Search.Data/preview/2019-05-06-preview/searchindex.json

```diff
@@ -1039,6 +1114,30 @@
       },
       "description": "Specifies the syntax of the search query. The default is 'simple'. Use 'full' if your query uses the Lucene query syntax."
     },
+    "FeaturesMode": {
+      "type": "string",
+      "enum": [
+        "disabled",
```

Comment:
is there a reason this is an enum of `disabled`\`enabled` rather than a simple boolean? Do we expect there to be a third state someday?

Reply:
Yes, we expect to extend this enum as we start adding new "features" through versioning. We will likely add new values such as maybe "enableV2" (not sure on the names yet)

Reply:
Not sure if it's already shipped on the server side, but maybe having it be "None", "V1", "V2" would make more semantic sense? But yeah, I get the difficulty here.

Reply:
Good suggestion. Unfortunately the preview API is already shipped and being used. However I'll bring that suggestion as we revisit the API for GA.

Reply:
AutoRest has extensions that allow you to change the name of a swagger enum but not the value you send out over the wire: https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-enum

If we think we can come up with better names, it would be great to change that sooner than later so we can expose them properly in our new client libraries.

I don't think we need to block this PR on getting that done now though.

Reply:
The thing is we aren't yet sure how the feature API will evolve. We published this as preview to get early feedback on it. We wanted to make the API flexible enough to allow us to use versioning, but aren't really ready to commit to it yet, that's why we started with enabled/disabled with the option to add more values to the enum in the future if needed.